src/smc.cpp

```diff
@@ -188,14 +200,50 @@ vec get_wgts(const umat &districts, int n_distr, int distr_ctr, bool final,
 
         lp[i] += add_constraint("splits", constraints,
             [&] (List l) -> double {
-                return eval_splits(districts.col(i), distr_ctr, counties, n_cty);
+                return eval_splits(districts.col(i), distr_ctr, as<uvec>(l["admin"]), l["n"]);
             });
 
         lp[i] += add_constraint("multisplits", constraints,
             [&] (List l) -> double {
-                return eval_multisplits(districts.col(i), distr_ctr, counties, n_cty);
+                return eval_multisplits(districts.col(i), distr_ctr, as<uvec>(l["admin"]), l["n"]);
             });
 
+        lp[i] += add_constraint("polsby", constraints,
+                                  [&] (List l) -> double {
+                                      return eval_polsby(districts.col(i), distr_ctr,
+                                                         as<ivec>(l["from"]),
+                                                         as<ivec>(l["to"]), as<vec>(l["area"]),
+                                                         as<vec>(l["perimeter"]));
+                                  });
+
+        lp[i] += add_constraint("fry_hold", constraints,
+                                  [&] (List l) -> double {
+                                      return eval_fry_hold(districts.col(i), distr_ctr,
+                                                           as<uvec>(l["total_pop"]),
+                                                           as<mat>(l["ssdmat"]),
+                                                           as<double>(l["denominator"]));
+                                  });
+
+        lp[i] += add_constraint("log_st", constraints,
```

Comment:
this one and the next one we should remove from SMC and M-S and just throw errors/warnings or something. I think it adds confusion & it's not good practice whatsoever. If they want one of these they should adjust the `compactness` argument.

The same might be said of `polsby` and `fry_hold` -- at minimum there should be some kind of message about turning off the default compactness if they want these.

Reply:
Yeah we can remove them and since it iterates by name, it won't break. The warning should probably be on the R end, I think?

I think the `polsby` one will have value here

Reply:
Sure, but there should be a message then, probably.  If they set compactness=0 and use polsby they could actually do OK efficiency-wise, but if they use both then it's gonna be bad

Reply:
Okay these are added now.